src/sound_openal.cpp

```diff
@@ -450,25 +450,29 @@ class OpenALSoundManager: public ISoundManager
 			const std::string &filepath)
 	{
 		SoundBuffer *buf = loadOggFile(filepath);
-		if(buf)
+		if (buf) {
 			addBuffer(name, buf);
-		return false;
+			return true;
+		} else {
+			return false;
+		}
 	}
 	bool loadSoundData(const std::string &name,
 			const std::string &filedata)
 	{
 		// The vorbis API sucks; just write it to a file and use vorbisfile
 		// TODO: Actually load it directly from memory
-		std::string basepath = porting::path_user + DIR_DELIM + "cache" +
-				DIR_DELIM + "tmp";
-		std::string path = basepath + DIR_DELIM + "tmp.ogg";
+		fs::CreateAllDirs(porting::path_cache);
+		std::string path = porting::path_cache + DIR_DELIM + "tmp.ogg";
 		verbosestream<<"OpenALSoundManager::loadSoundData(): Writing "
 				<<"temporary file to ["<<path<<"]"<<std::endl;
-		fs::CreateAllDirs(basepath);
 		std::ofstream of(path.c_str(), std::ios::binary);
 		of.write(filedata.c_str(), filedata.size());
 		of.close();
-		return loadSoundFile(name, path);
+		if (!loadSoundFile(name, path))
+			return false;
+		fs::DeleteSingleFileOrEmptyDirectory(path);
```

Comment:
Why only delete if loading the file succeeded? What if the problem was with the mod creator?